src/backend/package-stats/index.ts

```diff
@@ -0,0 +1,96 @@
+import { ComparisonOperator, Metric, MetricOptions, Statistic, TreatMissingData } from '@aws-cdk/aws-cloudwatch';
+import * as events from '@aws-cdk/aws-events';
+import * as targets from '@aws-cdk/aws-events-targets';
+import { IFunction, Tracing } from '@aws-cdk/aws-lambda';
+import { RetentionDays } from '@aws-cdk/aws-logs';
+import type { IBucket } from '@aws-cdk/aws-s3';
+import { Construct, Duration } from '@aws-cdk/core';
+import { lambdaFunctionUrl } from '../../deep-link';
+import { Monitoring } from '../../monitoring';
+import { RUNBOOK_URL } from '../../runbook-url';
+import { MetricName, METRICS_NAMESPACE } from './constants';
+import { PackageStats as Handler } from './package-stats';
+
+export interface PackageStatsProps {
+  /**
+   * The package store bucket, which should include both the
+   * catalog and stats.
+   */
+  readonly bucket: IBucket;
+
+  /**
+   * The monitoring handler to register alarms with.
+   */
+  readonly monitoring: Monitoring;
+
+  /**
+   * How long should execution logs be retained?
+   *
+   * @default RetentionDays.TEN_YEARS
+   */
+  readonly logRetention?: RetentionDays;
+}
+
+/**
+ * Builds or re-builds the `stats.json` object in the designated bucket.
+ */
+export class PackageStats extends Construct {
+  public readonly bucket: IBucket;
+  public readonly function: IFunction;
```

Comment:
The symbol `function` could be tricky somewheres:

```suggestion
  public readonly handler: IFunction;
```